pinot-core/src/main/java/org/apache/pinot/core/operator/docvalsets/ProjectionBlockValSet.java

```diff
@@ -52,6 +58,30 @@ public ProjectionBlockValSet(DataBlockCache dataBlockCache, String column, DataS
     _dataSource = dataSource;
   }
 
+  @Nullable
+  @Override
+  public RoaringBitmap getNullBitmap() {
+    if (!_nullBitmapSet) {
+      NullValueVectorReader nullValueReader = _dataSource.getNullValueVector();
+      ImmutableRoaringBitmap nullBitmap = nullValueReader != null ? nullValueReader.getNullBitmap() : null;
+      if (nullBitmap != null && nullBitmap.getCardinality() > 0) {
```

Comment:
For better performance, same for other places where cardinality is calculated just to identify if the bitmap is empty
```suggestion
      if (nullBitmap != null && !nullBitmap.isEmpty()) {
```